### PR TITLE
common-io: Refactoring SPI test cases

### PR DESCRIPTION
**Pull request by osloapps-max**
Description
-----------
Refactored tests to perform open/close during setup/tear-down phases.

This is part of the work related to the following issue: https://github.com/aws/amazon-freertos/issues/1859

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/common_io/test/test_iot_spi.c [hunk 1]
```diff
@@ -155,14 +173,7 @@ TEST_GROUP_RUNNER( TEST_IOT_SPI )
  */
 TEST( TEST_IOT_SPI, AFQP_IotSPI_OpenClose )
```
Comment (lundinc2) [inline libraries/abstractions/common_io/test/test_iot_spi.c:174]: Can delete this, since as you say it gets checked in setup/tear down

**Comment by osloapps-max** [comment]
@qiutongs I made the revert as requested. Also did some minor refactoring where I moved all code that could potentially fail between open/close calls into the TEST_PROTECT statement. Please let me know what you think about this.

**Comment by qiutongs** [comment]
> @qiutongs I made the revert as requested. Also did some minor refactoring where I moved all code that could potentially fail between open/close calls into the TEST_PROTECT statement. Please let me know what you think about this.

It looks great. Thank you for improving the test code quality!

**Comment by qiutongs** [comment]
One last thing is to format the code. Our team uses the [uncrustify tool](http://uncrustify.sourceforge.net/)

Please note to use this config: https://github.com/aws/amazon-freertos/blob/master/tools/uncrustify.cfg

**Pull request merged**